bad_code.java

```diff
@@ -0,0 +1,54 @@
+public class BadCodeExample {
+    
+    public static void main(String[] args) {
+        ChildClass child = new ChildClass();
+        child.doSomething();
+        child.doSomethingElse();
+    }
+}
+
+class ParentClass {
+    public int someNumber = 10;
+    public String someString = "Hello";
+
+    public void doSomething() {
+        System.out.println("Doing something in ParentClass");
+        someNumber++;
+        someString += " Worlds";
+        System.out.println(someString);
+    }
+
+    public void doSomethingElse() {
+        for (int i = 0; i < someNumber; i++) {
+            System.out.println("Count: " + i);
+            if (i % 2 == 0) {
+                System.out.println("Even number");
```

Comment:
To avoid repetitive code and enhance maintainability, consider creating a separate method to handle the printing of even and odd numbers, and call this method in both `doSomethingElse` methods of `ParentClass` and `ChildClass`. [medium]